### PR TITLE
Hook: useBackdropClick

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Isolates `useBackropClick` hook. 

Used in `SearchInput` and `Tooltip`

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: .changeset/fair-years-train.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/hooks': minor
+---
+
+Adds `useBackdropClick` hook. Fires a callback when any element except the provided refs is clicked
```
Comment (brookescarlett) [inline .changeset/fair-years-train.md:5]: nit: ref, singular

Comment (brookescarlett) [reply .changeset/fair-years-train.md:5]: or if plural, ref(s) are

File: packages/hooks/src/useBackdropClick.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  /**
+   * Whether the callback is enabled.
+   * It's recommended to set this to `false` when not in use,
+   * and toggle to `true` when the main elements (menu, tooltip, etc) are visible
+   */
+  enabled = true,
+): void {
+  /**
+   * We add two event handlers to the document to handle the backdrop click behavior.
+   * Intended behavior is to close the menu, and keep focus on the component.
```
Comment (shaneeza) [inline packages/hooks/src/useBackdropClick.ts:32]: Are these comments correct? This seems specific to a menu.

Comment (TheSonOfThomp) [reply packages/hooks/src/useBackdropClick.ts:32]: Ah, yeah I ported this over from menu, Thought I had caught all of these

File: .changeset/fair-years-train.md (hunk 1 above)
Comment (TheSonOfThomp) [reply .changeset/fair-years-train.md]: Updated!

File: packages/hooks/src/useBackdropClick.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  useEventListener(
+    'click',
+    (click: MouseEvent) => {
+      if (!doesComponentContainEventTarget(click)) {
+        click.stopPropagation(); // Stop any other click events from firing
+        callback();
+      }
+    },
+    {
+      options: { capture: true },
```
Comment (shaneeza) [inline packages/hooks/src/useBackdropClick.ts:67]: What is the benefit of handling this on capture? I'm not fully grasping the explanation above.

Comment (TheSonOfThomp) [reply packages/hooks/src/useBackdropClick.ts:67]: These handlers are on the body element, so firing the handler on `capture` means that it fires first, before the event propagates down to the deepest element

Comment (shaneeza) [reply packages/hooks/src/useBackdropClick.ts:67]: That makes sense!

**Review by shaneeza — APPROVED** [review]

**Pull request merged**